jobs/ror/ProcessRorCsv.php

```diff
@@ -0,0 +1,287 @@
+                "Importing batch chunk {$this->startRow}-{$this->endRow} failed: {$e->getMessage()}",
+                $this->scheduledTaskLogFilePath,
+                ScheduledTaskHelper::SCHEDULED_TASK_MESSAGE_TYPE_ERROR
+            );
+
+            $this->dropTemporaryTable();
+
+            throw $e;
+        }
+    }
+
+    /**
+     * Handle a job failure.
+     */
+    public function failed(?Throwable $exception): void
+    {
+        $this->dropTemporaryTable();
+
+        UpdateRorRegistryDataset::writeToExecutionLogFile(
+            "Importing batch chunk job has failed : {$exception->getMessage()}",
+            $this->scheduledTaskLogFilePath,
+            ScheduledTaskHelper::SCHEDULED_TASK_MESSAGE_TYPE_ERROR
+        );
+    }
+
+    /**
+     * Process the CSV row to generate the ROR data array.
+     */
+    protected function processRow(array $row): array
+    {
+        $ror = $row[$this->dataMappingIndex['ror']];
+        $displayLocale = (!empty($row[$this->dataMappingIndex['displayLocale']]))
+            ? $row[$this->dataMappingIndex['displayLocale']]
+            : $this->noLocale;
+        $isActive = (strtolower($row[$this->dataMappingIndex['isActive']]) === 'active') ? 1 : 0;
+        $searchPhrase = $ror;
+
+        $namesIn = $row[$this->dataMappingIndex['names']];
+        $namesOut = [];
+        if (!empty($namesIn)) {
+            $names = array_map('trim', explode(';', $namesIn));
+            for ($i = 0; $i < count($names); $i++) {
+                $name = array_map('trim', explode(':', $names[$i]));
+                if (count($name) === 2) {
+                    $namesOut[$name[0]] = trim($name[1]);
+                    $searchPhrase .= ' ' . $namesOut[$name[0]];
+                }
+            }
+        }
+
+        if (empty($namesOut[$displayLocale])) {
+            $namesOut[$displayLocale] = $row[$this->dataMappingIndex['displayName']];
+        }
+
+        return [
+            'ror' => $ror,
+            'displayLocale' => $displayLocale,
+            'isActive' => $isActive,
+            'searchPhrase' => trim($searchPhrase),
+            'name' => $namesOut,
+        ];
+    }
+
+    /**
+     * Process the CSV row to store in DB
+     */
+    protected function processBatch(array $rows): void
+    {
+        try {
+            $values = [];
+            
+            foreach ($rows as $row) {
+                foreach ($row['name'] as $locale => $name) {
+                    $values[] = [
+                        'ror' => $row['ror'],
+                        'display_locale' => $row['displayLocale'],
+                        'is_active' => $row['isActive'],
+                        'search_phrase' => $row['searchPhrase'],
+                        'locale' => $locale,
+                        'setting_name' => 'name',
+                        'setting_value' => $name
+                    ];
+                }
+            }
+            DB::table($this->temporaryTable)->insert($values);
```

Comment:
The temporary table has a unique key, which I guess will fail if a duplicated row is found here.